webpack/components/extensions/HostDetails/Tabs/PackagesTab.js

```diff
@@ -73,20 +91,157 @@ export const PackagesTab = () => {
   const response = useSelector(state => selectAPIResponse(state, HOST_PACKAGES_KEY));
   const { results, ...metadata } = response;
   const status = useSelector(state => selectHostPackagesStatus(state));
+  const {
+    selectOne,
+    isSelected,
+    searchQuery,
+    selectedCount,
+    isSelectable,
+    selectedResults,
+    updateSearchQuery,
+    selectNone,
+    selectAllMode,
+    areAllRowsSelected,
+    fetchBulkParams,
+    ...selectAll
+  } = useBulkSelect({
+    results,
+    metadata,
+  });
 
   if (!hostId) return <Skeleton />;
+
   const handleInstallPackagesClick = () => {
     setIsBulkActionOpen(false);
     setIsModalOpen(true);
   };
 
-  const rowActions = [
-    {
-      title: __('Upgrade via remote execution'), disabled: true,
-    },
-    {
-      title: __('Upgrade via customized remote execution'), disabled: true,
-    },
+  const removePackageViaRemoteExecution = packageName => dispatch(removePackage({
+    hostname,
+    packageName,
+  }));
+
+  const removeViaKatelloAgent = (packageName) => {
+    dispatch(removePackageViaKatelloAgent(hostId, { packages: [packageName] }));
+    selectNone();
+  };
+
+  const removePackagesViaRemoteExecution = () => {
+    const selected = fetchBulkParams();
+    setIsBulkActionOpen(false);
+    selectNone();
+    dispatch(removePackages({ hostname, search: selected }));
+  };
+
+  const selectedPackageNames = () => selectedResults.map(({ name }) => name);
+  const selectedUpgradableVersions = () => selectedResults.map(({ upgradable_version: v }) => v);
+
+  const removePackagesViaKatelloAgent = () => {
+    dispatch(removePackageViaKatelloAgent(hostId, { packages: selectedPackageNames() }));
+    selectNone();
+  };
+
+  const defaultRemoteAction = defaultRemoteActionMethod({ hostDetails });
+
+  const removeBulk = () => {
+    if (defaultRemoteAction === KATELLO_AGENT) {
+      removePackagesViaKatelloAgent();
+    } else {
+      removePackagesViaRemoteExecution();
+    }
+  };
+
+  const handlePackageRemove = (packageName) => {
+    if (defaultRemoteAction === KATELLO_AGENT) {
+      removeViaKatelloAgent(packageName);
+    } else {
+      removePackageViaRemoteExecution(packageName);
+    }
+  };
+
+  const upgradeViaRemoteExecution = packageName => dispatch(updatePackage({
+    hostname,
+    packageName,
+  }));
+
+  const upgradeBulkViaRemoteExecution = () => {
+    const selected = fetchBulkParams();
+    setIsBulkActionOpen(false);
+    selectNone();
+    dispatch(updatePackages({
+      hostname,
+      search: selected,
+    }));
+  };
+
+  const upgradeBulkViaKatelloAgent = () => {
+    if (areAllRowsSelected()) {
+      dispatch(upgradeAllViaKatelloAgent(hostId));
+    } else {
+      dispatch(upgradePackageViaKatelloAgent(hostId, { packages: selectedUpgradableVersions() }));
+    }
+    selectNone();
+  };
+
+  const upgradeBulk = () => {
+    if (!selectAllMode && defaultRemoteAction === KATELLO_AGENT) {
```

Comment:
The other code changes look good, but I think this still needs to be addressed?
```suggestion
    if (defaultRemoteAction === KATELLO_AGENT) {
```
I tested it and without this change, it still tries to use remote execution if I select all and click Upgrade.